### ISSUE TITLE
Un-released image version

### ISSUE THREAD

**Issue by Kolar23** [actions/runner-images#10224]
### Description

Hi there!

So I came to a very interesting discovery. I have a terraform repository, with different steps in the pipeline like apply, plan and etc. Out of nowhere my pipeline fails, saying that the plan was created in a different Terraform version, which got me confused. Upon looking closely in the pipeline, I saw the following:
![image](https://github.com/user-attachments/assets/1e4736b4-b693-4b58-b926-d1ed2a1476c2)

How is this even possible when the latest image version is 20240708 and it is used on my other steps of the pipeline. (This is the reason the pipeline fails)

### Platforms affected

- [X] Azure DevOps
- [ ] GitHub Actions - Standard Runners
- [ ] GitHub Actions - Larger Runners

### Runner images affected

- [ ] Ubuntu 20.04
- [X] Ubuntu 22.04
- [ ] Ubuntu 24.04
- [ ] macOS 12
- [ ] macOS 13
- [ ] macOS 13 Arm64
- [ ] macOS 14
- [ ] macOS 14 Arm64
- [ ] Windows Server 2019
- [ ] Windows Server 2022

### Image version and build link

Image version plan stage:
![image](https://github.com/user-attachments/assets/ad24d8a0-a86e-4ee3-9b53-d505aceff6c1)

Image version apply stage:
![image](https://github.com/user-attachments/assets/dd88b89f-8ed8-4887-b484-5a377c6fb7a6)

### Is it regression?

For now it happened only once, pipeline wasn't ran after that

### Expected behavior

Shouldn't be this version from what I'm seeing in the documentation.

### Actual behavior

Does the Terraform Plan with version 1.9.1 (the one in 20240708 image) and then the apply stage in Terraform 1.9.2 (the one in 20240711)

### Repro steps

Agent pool was already assigned, pipeline just ran

**Comment by Kolar23** [comment]
Found there is another issue raised with the same problem, regarding image versions. Closing the ticket.
